knowledge_repo/app/models.py

```diff
@@ -350,8 +350,18 @@ def groups(self, groups):
 
         # create an implicit group, group_post.id, to add
         # single users to
-        group = Group(name="group_" + self.id)
+        # we don't have self.id here necessarily, so query the Post table
+        # for the path
+        post_id = (db_session.query(Post)
+                             .filter(Post.path == self.path)
+                             .first()).id
+        group = Group(name="group_" + str(post_id))
```

Comment:
Do you think we should only create these implicit groups on demand? This creates the implicit post-specific group even if it will always be empty. Also, should we call the group something which cannot conflict with the set of other group names? Like... ':post_<id>', and then prevent the colon character from being used otherwise? I suggest it now only so we don't need to do migrations later if we do decided to do it cleanly in this way.